### PR TITLE
Refactor MIMXRT105x target labels and linker script, locate general data in DTCM memory instead of OCRAM for improved performance

### PR DESCRIPTION
**Pull request by multiplemonomials**
### Summary of changes 
On the MIMXRT1050/1060 chips, the DTCM memory is significantly faster than the OCRAM memory.  Sometimes the CPU cache can help with this, but the DTCM memory bypasses the cache entirely so its full size can be accessed at full speed.  

Therefore, it makes sense to move as much stuff into DTCM as possible, especially on devices without an external RAM chip such as the Teensy boards.  This PR does that, and also makes a refactor to the linker script and the target labels for MIMXRT105x to make this possible.

In particular, this PR adds a new target label, `MIMXRT105x`, to indicate code for BOTH the 1050-series and the 1060-series chips.  Code specific to the 1050 can now be guarded with `MBED_TARGET_MIMXRT1050`, and code specific to the 1060 can be guarded with `MBED_TARGET_MIMXRT1060`.

#### Impact of changes 
- Execution speed should be moderately increased for Teensy boards.
- TARGET_MIMXRT1050 label changed from referring to both the 1050 and 1060 to the 1050 only

#### Migration actions required 
- If user code was using the TARGET_MIMXRT1050 label anywhere, it should change to `TARGET_MIMXRT105x` for the same behavior.  I don't think this is likely to be used anywhere though.

### Documentation 

None

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [X] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT105x/TARGET_EVK/CMakeLists.txt [hunk 1]
```diff
@@ -34,7 +34,6 @@ target_include_directories(mbed-mimxrt1060-evk
 
 target_sources(mbed-mimxrt1060-evk
     INTERFACE
-        fsl_phy.c
```
Comment (multiplemonomials) [inline targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_MIMXRT105x/TARGET_EVK/CMakeLists.txt:37]: Fixes compile error for MIMXRT1060_EVK, whoops

File: tools/cmake/upload_methods/UploadMethodJLINK.cmake [hunk 2]
```diff
@@ -61,7 +61,7 @@ exit
 		COMMAND ${JLINK}
 		${JLINK_SELECT_ARG}
 		${JLINK_NOGUI_ARG}
-		-Device ${JLINK_CPU_NAME}
+		-Device \"${JLINK_CPU_NAME}\"
```
Comment (multiplemonomials) [inline tools/cmake/upload_methods/UploadMethodJLINK.cmake:64]: Fixes a bug where the loader selection arg to J-Link was missing quotes so it didn't work

**Comment by multiplemonomials** [comment]
@wwatson4506 this look OK?

**Comment by multiplemonomials** [comment]
MIMXRT1060_EVK can never pass the pin validation checks due to how the dev board is physically wired.  So, merging despite CI failure.

**Comment by wwatson4506** [comment]
That is what I was seeing was the pin check failure.

**Comment by multiplemonomials** [comment]
yeah unfortunately, they wired the same physical pin to multiple pins on the Arduino header, which fails the CI job which validates all the Arduino pins are in compliance with a standard mapping

**Pull request merged**